src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -728,6 +728,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             if (hasNaturalType)
             {
                 tupleTypeOpt = TupleTypeSymbol.Create(node.Location, elements, locations, elementNamesArray, this.Compilation, node, diagnostics);
+                Binder.ReportDiagnosticsIfObsoleteInternal(diagnostics, tupleTypeOpt, node, tupleTypeOpt.ContainingType, BinderFlags.None);
```

Comment:
Will this handle nested case properly? Does it make sense to do this inside TupleTypeSymbol.Create?